build-system/pr-check.js

```diff
@@ -376,7 +377,8 @@ function main(argv) {
       command.runDepAndTypeChecks();
       // Skip unit tests if the PR only contains changes to integration tests.
       if (buildTargets.has('RUNTIME')) {
-        command.runUnitTests();
+        command.buildValidator();
```

Comment:
If this is run, line 388 will be duplicated work, and will add an extra 2m 30s to each build, increasing the total Travis CPU time by about 5 minutes (see my other comment for line 298). At a minimum, we'd need to change the `if` case in line 387 to `buildTargets.has('VALIDATOR') && !buildTargets.has('RUNTIME') `